### PR TITLE
Many improvements

### PR DESCRIPTION
**Pull request by gagern**
I very much like the core idea of your project, but I find the actual code to be lacking. In my own branch I've addressed numerous issues, and decided against filing a bug report for each of them. Please have a look at the commits in this pull request, since I outlined the issue in question in each of the commit messages.

There are some things which I haven't addressed but which I have my doubts about. One is the fact that you wrapped the bodies of many functions in try/catch blocks. That won't do any good for the nested functions therein, but it makes synchroneous errors at the top level asynchroneous as well. I'm not sure I consider this a good thing; I'd leave those exceptions alone and use rejected promises only for asynchroneous error reporting. Was there a specific reason why you introduced those blocks e.g. in 7b342f5069e19b02aa0d4f0b49cd0cac59526dda?

Another thing that's somewhat unclean: in [the startup detection code](https://github.com/hiroaki-yamamoto/node-validator-nu/blob/7b342f5069e19b02aa0d4f0b49cd0cac59526dda/src/remote.coffee#L26) you expect to read the “server started” message in a single chunk. Due to unfortunate buffering, however, it might be that you receive this line across different chunks. So theoretically you'd have to buffer and concatenate all input until you have that line. I did implement the buffering (for better error reporting on startup failure) but haven't used it for that purpose yet.

To be honest, I've got half a mind to write all of this anew from scratch. Partly because I don't feel at home with Coffeescript. I'll see how usable this package is now, and decide after I gained some more experience with it.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
-node_modules
+node_modules/
+/lib/
```
Comment (hiroaki-yamamoto) [inline .gitignore]: lib dir is needed. How can I contain the compiled file??

File: Gulpfile.coffee [hunk 2]
```diff
@@ -25,7 +25,7 @@ gulp.task "test", ["stylecheck"], ->
   gulp.src(
     "tests/spec/*.coffee"
   ).pipe(mocha(
-    "timeout": 10000
+    "timeout": 30000
```
Comment (hiroaki-yamamoto) [inline Gulpfile.coffee]: Too long timeout.

**Comment by hiroaki-yamamoto** [comment]
Hi :smile:
Sorry, I don't have enough time to review for now, but I will be back for review.

# This PR seems to be pretty awesome except some points.

File: .gitignore (hunk 1 above)
Comment (gagern) [inline .gitignore]: See 0ca7cc9aebee196f9e9f81dc32e9d4afa29df0e1 for context and explanation. The directory is removed from git, but included in the published package, and the `prepublish` script will ensure that it's up to date when you publish. If you want to give this a dry run, make `npm pack` and you'll get a tarball of everything that will get published.

File: Gulpfile.coffee (hunk 2 above)
Comment (gagern) [inline Gulpfile.coffee]: Well, with the 10000 timeout, Travis [ran into a timeout](https://travis-ci.org/hiroaki-yamamoto/node-validator-nu/builds/79910968). Might be that 15000 would be enough, but on the other hand, what's the problem with this timeout? You *have* to expect JVM startup to take considerable time on many architectures, and the way I understand it, v.nu gets some schema files from the web, which will take some time as well, depending on the internet connection. I'd rather play this safe, and perhaps [explicitely reduce timeouts](http://mochajs.org/#timeouts) for the `remote` suite which should be faster if there is any real reason for doing so.

**Comment by gagern** [comment]
I suggest you review this one commit at a time, not over the whole diff, since the commit messages usually give some explanation of what I'm doing.

**Comment by hiroaki-yamamoto** [comment]
Ok, thanks I will merge and make some changes after I finished gulp-scss issue.

**Pull request merged**